ClientApp/viking-vault/src/Dashboard.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import {constants} from "./ConstantVariables";
+import { Redirect } from "react-router-dom";
+import UserPage from "./components/UserPage";
+import { AdminPage } from "./display_user_list/AdminPage";
+
+const baseUrl = constants.baseUrl;
+
+interface IState {
+    isAdmin: boolean | null;
+}
+
+class Dashboard extends React.Component<any, IState> {
+    state = {
+        isAdmin: null
```

Comment:
A variable that begins with isAdmin suggest that is a Boolean variable. Maybe it's a good idea to initialize it with false instead of null

Reply:
Null is like an intermediate state. It is needed to know that the response from the server hasn’t arrived yet. It can’t be initialized with false because that would mean that we assume that it’s a simple user before we get to ask the server